internal/data/health/health.go

```diff
@@ -22,29 +22,24 @@ func New(g *gorm.DB, a authzapi.Authorizer) *healthRepo {
 
 func (r *healthRepo) IsBackendAvailable(ctx context.Context) (*pb.GetReadyzResponse, error) {
 	storageType := r.DB.Dialector.Name()
-	log.Infof("Checking readiness for storage type: %s", storageType)
-
-	sqlDB, err := r.DB.DB()
-	if err != nil {
-		log.Errorf("Failed to retrieve DB: %v", err)
-		return newResponse("STORAGE UNHEALTHY: "+storageType, 500), nil
+	sqlDB, dbErr := r.DB.DB()
+	if dbErr == nil {
+		dbErr = sqlDB.PingContext(ctx)
 	}
-	if err := sqlDB.PingContext(ctx); err != nil {
-		log.Errorf("Failed to ping database: %v", err)
-		return newResponse("FAILED TO PING DB: "+storageType, 500), nil
-	} else {
-		log.Infof("Successfully pinged %s database", storageType)
-	}
-	// if kessel is enabled
-	health, err := r.Authz.Health(ctx)
-	log.Infof("health status %s error status %s", health, err)
-	if err != nil {
-		log.Infof("Checking readiness of relations-api")
+	health, apiErr := r.Authz.Health(ctx)
 
-		//log.Infof("Storage type %s and relations API are healthy", storageType)
-		return newResponse("KESSEL UNHEALTHY:"+health.GetStatus(), 500), nil
+	if dbErr != nil && apiErr != nil {
```

Comment:
If `dbErr` is not nil but `apiErr` is nil than `&&` condition won't return `dbErr` as `"STORAGE UNHEALTHY: "+storageType+"`